models/repo_list.go

```diff
@@ -146,6 +148,10 @@ type SearchRepoOptions struct {
 	// True -> include just archived
 	// False -> include just non-archived
 	Archived util.OptionalBool
+	// None -> include pinned AND non-pinned
+	// True -> include just pinned
+	// False -> include just non-pinned
+	Pinned util.OptionalBool
```

Comment:
Interesting. I wouldn't have thought about offering the ability to search by `pinned` or `unpinned`.
I don't see a usecase for that.
Apparently we have completely differing opinions on what the effect of pinning a repository means:
For me, a pinned repo is simply a stylistic element so that the most important repositories can be quickly found by visiting the repo list of the user.
For you, a pinned repo is apparently additionally a state a repo can have to signify its importance.

Reply:
@delvh I'm on the same opinion as you about what pinned repos are, however I think this may be useful if we kid the page eventually via JS/API and we could filter based on pinned repos to display (vs having to do that filtering client side)

Reply:
If I didn't do this, how would you suggest I extract pinned repositories for the repo listings?

Currently I'm doing two searches, one for pinned and one for unpinned and then merging them. Would you have me fetch them all and then manually reorder the pinned repositories to the front?

Reply:
I would have followed the GitHub approach: Simply show the pinned repositories twice, once above everything else, and once inside the list of **all** repositories.